dummyserver/server.py

```diff
@@ -16,6 +16,11 @@
 import warnings
 from datetime import datetime
 
+try:
+    from ssl import CERT_OPTIONAL
+except:
+    CERT_OPTIONAL = None
```

Comment:
... :man_facepalming: I'll be honest, I didn't review more of the test code than what was necessary to fix the tests that were broken.  I'll go back and take a closer look at this.